perllib/FixMyStreet/App/Controller/Questionnaire.pm

```diff
@@ -151,12 +151,23 @@ sub record_state_change : Private {
     my $problem = $c->stash->{problem};
     my $old_state = $c->stash->{old_state};
     my $new_state = '';
-    $new_state = 'fixed - user' if $c->stash->{been_fixed} eq 'Yes' &&
-        FixMyStreet::DB::Result::Problem->open_states()->{$old_state};
-    $new_state = 'fixed - user' if $c->stash->{been_fixed} eq 'Yes' &&
-        FixMyStreet::DB::Result::Problem->closed_states()->{$old_state};
-    $new_state = 'confirmed' if $c->stash->{been_fixed} eq 'No' &&
-        FixMyStreet::DB::Result::Problem->fixed_states()->{$old_state};
+    if ($c->cobrand->call_hook("prevent_questionnaire_updating_status")) {
+        $new_state = '';
+        if ($c->stash->{been_fixed} eq 'Yes') {
+            $c->stash->{status_comment} = 'report is fixed';
+        } elsif ($c->stash->{been_fixed} eq 'No') {
+            $c->stash->{status_comment} = 'report is not fixed';
+        } else {
+            $c->stash->{status_comment} = 'they don\'t know if report is fixed'
+        }
+    } else {
+        $new_state = 'fixed - user' if $c->stash->{been_fixed} eq 'Yes' &&
+            FixMyStreet::DB::Result::Problem->open_states()->{$old_state};
+        $new_state = 'fixed - user' if $c->stash->{been_fixed} eq 'Yes' &&
+            FixMyStreet::DB::Result::Problem->closed_states()->{$old_state};
+        $new_state = 'confirmed' if $c->stash->{been_fixed} eq 'No' &&
+            FixMyStreet::DB::Result::Problem->fixed_states()->{$old_state};
+    }
     $c->stash->{new_state} = $new_state;
 
     # Record state change, if there was one
```

Comment:
You still have new_state set to blank, so the actual thing they submitted is not saved in the questionnaire (as it would be in line 190 below), but instead something they did not put is potentially saved.